qemu/tests/virtio_fs_daemon_readonly.py

```diff
@@ -0,0 +1,186 @@
+import re
+
+from avocado.utils import process
+from virttest import error_context, utils_disk, utils_misc, utils_test
+from virttest.utils_version import VersionInterval
+
+from provider import virtio_fs_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test virtio-fs daemon exported with --read-only option.
+    Steps:
+        1. Create a shared directory and a file for testing on the host.
+        2. Run the virtiofsd daemon with --readonly on the host.
+        3. Boot a guest on the host.
+        4. Log into guest then mount the virtiofs.
+        5. Test the mount dir is readonly.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def create_service(session):
+        if os_type == "windows":
+            error_context.context("Create virtiofs service in guest.", test.log.info)
+
+            driver_name = params["driver_name"]
+
+            session = utils_test.qemu.windrv_check_running_verifier(
+                session, vm, test, driver_name
+            )
+            viofs_svc_name = params["viofs_svc_name"]
+            virtio_fs_utils.create_viofs_service(
+                test, params, session, service=viofs_svc_name
+            )
+        return session
+
+    def delete_service():
+        if os_type == "windows":
+            error_context.context("Delete virtiofs service in guest.", test.log.info)
+            session = vm.wait_for_login()
+            virtio_fs_utils.delete_viofs_serivce(test, params, session)
+            session.close()
+
+    def start_service(session):
+        def start_multifs_instance(fs_tag, fs_target, fs_volume_label):
```

Comment:
There is no multifs definition in the cfg file, and multifs seems not enforced.  How about removing it.

Reply:
Updated. thanks. And please help to take a review again.